run/o2sim_parallel.cxx

```diff
@@ -89,6 +91,24 @@ void remove_tmp_files()
 
 void cleanup()
 {
+  auto& conf = o2::conf::SimConfig::Instance();
+  if (conf.forwardKine()) {
+    auto factory = fair::mq::TransportFactory::CreateTransportFactory("zeromq");
+    auto forwardchannel = fair::mq::Channel{"kineforward", "pair", factory};
+    auto address = std::string{"ipc:///tmp/o2sim-hitmerger-kineforward-"} + std::to_string(getpid());
+    forwardchannel.Bind(address.c_str());
+    forwardchannel.Validate();
+    fair::mq::Parts parts;
+    fair::mq::MessagePtr payload(forwardchannel.NewMessage());
+    o2::framework::SourceInfoHeader sih;
+    sih.state = o2::framework::InputChannelState::Completed;
+    auto channelAlloc = o2::pmr::getTransportAllocator(forwardchannel.Transport());
+    auto header = o2::pmr::getMessage(o2::header::Stack{channelAlloc, sih});
+    parts.AddPart(std::move(header));
+    parts.AddPart(std::move(payload));
+    forwardchannel.Send(parts);
+    LOGP(info, "SENDING END-OF-STREAM TO PROXY AT {}", address.c_str());
+  }
```

Comment:
couldn't we have this in the HitMerger (which also knows when all
events are done)? Or do you target EOS even in case of failures here?

Reply:
Indeed, this way it will be sent even in case of failures so the proxy will not be stuck if o2sim was killed. We can also potentially avoid event counting in the proxy, but I did not touch that for now.